### PR TITLE
protect against missing backups in manifests

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

If a prior manifest is missing a backup, it should be handled the same way as when a manifest's backup is missing a details ID: the metadata is skipped, and a full backup is performed.

## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :bug: Bugfix

## Issue(s)

* #2062

## Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/operations/backup.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ func (op *BackupOperation) Run(ctx context.Context) (err error) {
 		}
 	}()
 
-	mans, mdColls, err := produceManifestsAndMetadata(ctx, op.kopia, op.store, oc, tenantID, uib)
+	mans, mdColls, canUseMetaData, err := produceManifestsAndMetadata(ctx, op.kopia, op.store, oc, tenantID, uib)
```
Comment (ashmrtn) [inline src/internal/operations/backup.go:154]: this works for now, but we're probably going to want something more granular in the future

File: src/internal/operations/backup.go [hunk 2]
```diff
@@ -344,12 +344,22 @@ func produceManifestsAndMetadata(
 
 		bID := man.Tags[tk]
 		if len(bID) == 0 {
-			return nil, nil, errors.New("missing backup id in prior manifest")
+			return nil, nil, false, errors.New("manifest missing backup ID")
```
Comment (ashmrtn) [inline src/internal/operations/backup.go:347]:
```suggestion
			return nil, nil, false, errors.New("snapshot manifest missing backup ID")
```

**Review by ashmrtn — APPROVED** [review]
can you make tickets for the following please:
* unit tests for `produceManifestsAndMetadata`
* fine-grained control over which bases are used for delta token-based incrementals so that we be as efficient as possible

File: src/internal/operations/backup.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/operations/backup.go:154]: agreed

**Pull request merged**